rules/microsoft_rules/microsoft365_external_sharing.py

```diff
@@ -0,0 +1,44 @@
+import re
+from fnmatch import fnmatch
+
+from panther_base_helpers import m365_alert_context
+
+email_regex = re.compile(r"([A-Za-z0-9]+[.-_])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\.[A-Z|a-z]{2,})+")
+
+ALLOWED_DOMAINS = ["mycompany.com", "alloweddomain.com"]
+
+ALLOWED_USERS = ["exception@outsider.com"]
```

Comment:
We should leave a comment that `ALLOWED_USERS` should be lowercased only, per line29.